packages/adapters/solana/src/client.ts

```diff
@@ -477,14 +477,14 @@ export class SolanaAdapter implements ChainAdapter {
         provider.chains.find(chain => chain.caipNetworkId === caipNetworkId) || provider.chains[0]
 
       if (connectionChain) {
+        ProviderUtil.setProvider(this.chainNamespace, provider)
+        this.provider = provider
+
```

Comment:
why was this needed?

Reply:
It was a leftover attempt to keep states synced as the provider was being get as undefined at some point due to the async task on network switch. I'm going to revert it.